diff --git a/node_modules/react-native-paper/lib/module/components/Avatar/AvatarImage.js b/node_modules/react-native-paper/lib/module/components/Avatar/AvatarImage.js
index aa97826..a0c6136 100644
--- a/node_modules/react-native-paper/lib/module/components/Avatar/AvatarImage.js
+++ b/node_modules/react-native-paper/lib/module/components/Avatar/AvatarImage.js
@@ -30,6 +30,8 @@ const AvatarImage = ({
   source,
   style,
   theme,
+  onError,
+  onLoad,
   ...rest
 }) => {
   const {
@@ -53,7 +55,9 @@ const AvatarImage = ({
       width: size,
       height: size,
       borderRadius: size / 2
-    }
+    },
+    onError: onError,
+    onLoad: onLoad
   }));
 };
 
diff --git a/node_modules/react-native-paper/lib/module/components/Drawer/DrawerItem.js b/node_modules/react-native-paper/lib/module/components/Drawer/DrawerItem.js
index a5137b3..0839f26 100644
--- a/node_modules/react-native-paper/lib/module/components/Drawer/DrawerItem.js
+++ b/node_modules/react-native-paper/lib/module/components/Drawer/DrawerItem.js
@@ -7,6 +7,7 @@ import Text from '../Typography/Text';
 import Icon from '../Icon';
 import TouchableRipple from '../TouchableRipple/TouchableRipple';
 import { withTheme } from '../../core/theming';
+import Badge from '../Badge';
 
 /**
  * A component used to show an action item with an icon and a label in a navigation drawer.
@@ -41,6 +42,7 @@ const DrawerItem = ({
   style,
   onPress,
   accessibilityLabel,
+  badge = 0,
   ...rest
 }) => {
   const {
@@ -71,6 +73,8 @@ const DrawerItem = ({
       selected: active
     },
     accessibilityLabel: accessibilityLabel
+  }, /*#__PURE__*/React.createElement(View, {
+    style: styles.badgeContainer
   }, /*#__PURE__*/React.createElement(View, {
     style: styles.wrapper
   }, icon ? /*#__PURE__*/React.createElement(Icon, {
@@ -85,7 +89,9 @@ const DrawerItem = ({
       ...font,
       marginLeft: labelMargin
     }]
-  }, label))));
+  }, label)), /*#__PURE__*/React.createElement(Badge, {
+    visible: Boolean(badge)
+  }, badge))));
 };
 
 DrawerItem.displayName = 'Drawer.Item';
@@ -101,6 +107,12 @@ const styles = StyleSheet.create({
   },
   label: {
     marginRight: 32
+  },
+  badgeContainer: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: 8
   }
 });
 export default withTheme(DrawerItem);
